components/camel-tahu/src/main/java/org/apache/camel/component/tahu/TahuEndpoint.java

```diff
@@ -0,0 +1,268 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.tahu;
+
+import java.util.Arrays;
+import java.util.List;
+
+// import java.util.Map;
+import org.apache.camel.Category;
+import org.apache.camel.Consumer;
+import org.apache.camel.FailedToCreateConsumerException;
+import org.apache.camel.FailedToCreateProducerException;
+import org.apache.camel.Processor;
+import org.apache.camel.Producer;
+import org.apache.camel.spi.HeaderFilterStrategy;
+import org.apache.camel.spi.HeaderFilterStrategyAware;
+import org.apache.camel.spi.Metadata;
+import org.apache.camel.spi.UriEndpoint;
+import org.apache.camel.spi.UriParam;
+import org.apache.camel.spi.UriPath;
+import org.apache.camel.support.DefaultEndpoint;
+import org.apache.camel.support.DefaultHeaderFilterStrategy;
+import org.apache.camel.util.ObjectHelper;
+import org.eclipse.tahu.message.BdSeqManager;
+import org.eclipse.tahu.message.model.SparkplugBPayloadMap;
+
+/**
+ * Sparkplug B Edge Node and Host Application support over MQTT using Eclipse Tahu
+ */
+@UriEndpoint(firstVersion = "4.0.0",
```

Comment:
4.8.0

Reply:
Guess I've been working on this for awhile....